low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,411 @@
+                self.failed_services.append(service)
+                alert_index = 1
+            else:
+                alert_index = 4
+        elif prev_state == "inactive" or prev_state == "failed":
+            if state == "activating":
+                if service not in self.not_active_services:
+                    self.not_active_services[service] = int(time.time())
+            elif state == "active":
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
+                    alert_index = 3
+                elif service in self.not_active_services:
+                    self.not_active_services.pop(service)
+            else:
+                alert_index = 4
+        elif prev_state == "activating":
+            if service in self.not_active_services:
+                self.not_active_services.pop(service)
+            if state == "inactive" or state == "deactivating":
+                self.failed_services.append(service)
+                alert_index = 1
+            elif state == "active":
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
+                    alert_index = 3
+                else :
+                    # its a restart.
+                    pass
+            elif state == "failed":
+                self.failed_services.append(service)
+                alert_index = 1
+            else:
+                alert_index = 4
+
+        logger.debug(f"alert_index : {alert_index}")
+        if not self.not_active_services:
+            logger.debug("Not_active_services is EMPTY")
+        for s,t in self.not_active_services.items():
+            logger.debug(f"{s} : {t}, {int(time.time())}")
+        logger.debug(f"{self.failed_services}")
+
+        if alert_index != -1:
+            self.raise_alert(service, prev_state, state,
+                prev_substate, substate, prev_pid, pid,
+                alert_index)
+
+    def raise_alert(self, service, prev_state, state, prev_substate,
+                    substate, prev_pid, pid, alert_index):
+        """ Send the alert to ServiceMsgHandler."""
+        # Each alert info contains 4 fields
+        # 1.Description | 2.Alert Type | 3.Impact | 4.Recommendation
+        alert_info = [
+            [f"{service} is not found/running initially",   #index 0
+                "fault", "", ""],
+            [f"{service} is in {state} state.",             #index 1
+                "fault", "", ""],
+            [f"{service} is in a non_active state for more than {self.max_wait_time} seconds.",
+                "fault", "", ""],                           #index 2
+            [f"{service} returned to running state.",
+                "fault_resolved", "", ""],                  #index 3
+            [f"service state transition from {prev_state} to {state} is not handled.",
+                "missing", "", ""]                          #index 4
```

Comment:
I was thinking it as a alert for sspl,  that this transition has occurred but we according to our logic has not handled it..  
Previously I searched on net and found that inactive to active is not possible so did not include it in the transition but rsyslog.service transitioned from inactive to active and I got an alert that inactive->active is not handled.. so this could be a good alert of sspl to improve  I think..